zerver/lib/zcommand.py

```diff
@@ -54,4 +58,47 @@ def change_mode_setting(command: str, switch_command: str,
                                             switch_command='fluid-width',
                                             setting='fluid_layout_width',
                                             setting_value=False))
+    elif command == 'digress':
+        data_keys = [
+            "old_stream",
+            "old_topic",
+            "new_stream",
+            "new_topic",
+        ]
+        if not data or not all(key in data for key in data_keys):
+            raise JsonableError(_('Invalid data.'))
+
+        (old_stream, ignored_old_stream_sub) = access_stream_by_name(
+            user_profile, data["old_stream"])
+
+        (new_stream, ignored_new_stream_sub) = access_stream_by_name(
+            user_profile, data["new_stream"])
+
+        old_topic_exists = user_message_exists_for_topic(
+            user_profile=user_profile,
+            recipient_id=old_stream.recipient_id,
+            topic_name=data["old_topic"],
+        )
+
+        if not old_topic_exists:
+            return dict(
+                msg=_('Old topic #**{}>{}** does not exist.').format(data["old_stream"], data["old_topic"])
+            )
+
+        if data["old_stream"] == data["new_stream"]:
+            if data["old_topic"] == data["new_topic"]:
```

Comment:
Why not do this in the same line? Also, since checking this doesn't require access to stream, we should move it next to 'Invalid data` check.

Reply:
I was worried about line length, but this didn't turn out to be any longer than some of the other lines. Updated this, and moved it up, along with it's test.